airflow-core/src/airflow/api_fastapi/core_api/services/public/connections.py

```diff
@@ -56,11 +59,24 @@ def update_orm_from_pydantic(
     if (not update_mask and "password" in pydantic_conn.model_fields_set) or (
         update_mask and "password" in update_mask
     ):
-        orm_conn.set_password(pydantic_conn.password)
+        if pydantic_conn.password is None:
+            orm_conn.set_password(pydantic_conn.password)
+            return
+
+        merged_password = merge(pydantic_conn.password, orm_conn.password, "password")
+        orm_conn.set_password(merged_password)
```

Comment:
```suggestion
        else:
            merged_password = merge(pydantic_conn.password, orm_conn.password, "password")
            orm_conn.set_password(merged_password)
```

Do we really want to return? What if we have no password but an extra?

Reply:
You're right, that's not correct, fixed.